packages/zwave-js/src/lib/commandclass/SceneActuatorConfigurationCC.ts

```diff
@@ -0,0 +1,395 @@
+import {
+	CommandClasses,
+	Duration,
+	Maybe,
+	MessageOrCCLogEntry,
+	validatePayload,
+	ValueID,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import {
+	CCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	SetValueImplementation,
+	SET_VALUE,
+	throwMissingPropertyKey,
+	throwUnsupportedProperty,
+	throwUnsupportedPropertyKey,
+	throwWrongValueType,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+
+// All the supported commands
+export enum SceneActuatorConfigurationCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+}
+
+export function getLevelValueID(
+	endpoint: number | undefined,
+	sceneId: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Scene Actuator Configuration"],
+		endpoint,
+		property: "level",
+		propertyKey: sceneId,
+	};
+}
+
+export function getDimmingDurationValueID(
+	endpoint: number | undefined,
+	sceneId: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Scene Actuator Configuration"],
+		endpoint,
+		property: "dimmingDuration",
+		propertyKey: sceneId,
+	};
+}
+
+function persistSceneActuatorConfig(
+	this: SceneActuatorConfigurationCC,
+	sceneId: number,
+	level: number,
+	dimmingDuration: Duration,
+) {
+	const levelValueId = getLevelValueID(this.endpointIndex, sceneId);
+	const dimmingDurationValueId = getDimmingDurationValueID(
+		this.endpointIndex,
+		sceneId,
+	);
+	const valueDB = this.getValueDB();
+
+	if (!valueDB.hasMetadata(levelValueId)) {
+		valueDB.setMetadata(levelValueId, {
+			...ValueMetadata.UInt8,
+			label: `Level (${sceneId})`,
+		});
+	}
+	if (!valueDB.hasMetadata(dimmingDurationValueId)) {
+		valueDB.setMetadata(dimmingDurationValueId, {
+			...ValueMetadata.Duration,
+			label: `Dimming duration (${sceneId})`,
+		});
+	}
+
+	valueDB.setValue(levelValueId, level);
+	valueDB.setValue(dimmingDurationValueId, dimmingDuration);
+
+	return true;
+}
+
+@API(CommandClasses["Scene Actuator Configuration"])
+export class SceneActuatorConfigurationCCAPI extends CCAPI {
+	public supportsCommand(
+		cmd: SceneActuatorConfigurationCommand,
+	): Maybe<boolean> {
+		switch (cmd) {
+			case SceneActuatorConfigurationCommand.Get:
+				return this.isSinglecast();
+			case SceneActuatorConfigurationCommand.Set:
+			case SceneActuatorConfigurationCommand.Report:
+				return true;
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
```

Comment:
The setValue API seems awkward for this CC. It is meant to enable a scene, optionally with a target level. But to enable the scene, one would have to set the `level` property for that scene. Maybe this should be left out.

Reply:
I thought this class was for configuring how the actuator should react when a scene is activated.  When we call a `Set`, we're telling the endpoint that when scene `sceneId` is activated, that actuator should take `dimmingDuration` amount of time to go to a target value of `level` (or optionally the `current level` of the device as of when we're calling the `SceneActuatorConfigurationCC.Set`).  In some sense setting the `level` of scene `sceneId` is the most fundamental part of the `Set` command, because this `level` is always used when the scene is activated (while the `dimmingDuration` assigned here can be overridden by a value from the node that sends the `SceneActivation.Set` command.

So, I think being able to use the setValue API to update the `level` assigned to the scene does make some sense.  However, I do think it's a bit awkward to allow setting of `level` but not allow setting of `dimmingDuration`.  Also, you have a much better understanding of the whole API.

Reply:
> I thought this class was for configuring how the actuator should react when a scene is activated.

You're right - I mixed it up while reading yesterday. Setting level is okay, setting the duration can also be done if you accept Duration objects. I was thinking of those values where the duration is a kind of meta value for another one, e.g. "go to level X with duration Y now!"
Same probably also goes for the SceneControllerConfigurationCC then?